internal/backends/postgresql/collection.go

```diff
@@ -245,8 +248,53 @@ func (c *collection) DeleteAll(ctx context.Context, params *backends.DeleteAllPa
 
 // Explain implements backends.Collection interface.
 func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams) (*backends.ExplainResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3389
-	return new(backends.ExplainResult), nil
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if p == nil {
+		return new(backends.ExplainResult), nil
+	}
+
+	meta, err := c.r.CollectionGet(ctx, c.dbName, c.name)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if meta == nil {
+		return &backends.ExplainResult{
+			QueryPlanner: must.NotFail(types.NewDocument()),
+		}, nil
+	}
+
+	// TODO https://github.com/FerretDB/FerretDB/issues/3414
+	q := fmt.Sprintf(
+		`EXPLAIN (VERBOSE true, FORMAT JSON) SELECT %s FROM %s`,
+		metadata.DefaultColumn,
+		pgx.Identifier{c.dbName, meta.TableName}.Sanitize(),
+	)
+
+	rows, err := p.Query(ctx, q)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	iter := newQueryIterator(ctx, &iteratorParams{
+		rows:      rows,
+		unmarshal: unmarshalExplain,
+	})
+
+	defer iter.Close()
+
+	_, queryPlan, err := iter.Next()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
```

Comment:
I meant simplifying as just query that one row we need and unmarshall without using query iterator. It's not necessary to iterate on one row. :hugs: 
```suggestion
	var b []byte
	err = p.QueryRow(ctx, q).Scan(&b)
	if err != nil {
		return nil, lazyerrors.Error(err)
	}

	queryPlan, err := unmarshalExplain(b)
	if err != nil {
		return nil, lazyerrors.Error(err)
	}
```